Source/items.h

```diff
@@ -348,6 +349,20 @@ struct ItemStruct {
 			return false;
 		}
 	}
+
+	UiFlags getTextColor() const

```

Comment:
I know this is a simple "move" refactor, but this whole method seems like a tell-dont-ask violation to me. Shouldn't we just have a "renderToText" function of some sort that does the rendering and then make the color decision an implementation detail?

One could even argue that having display logic in the item class is an SRP violation as well. Ideally, logic and display concerns should not be in the same class in the first place. Have you thought about creating a different abstraction to do put this logic in?

Reply:
I thought about it but it's extra work -- for a later refactor. It's a minor thing too because these colors are the same throughout all Diablo games and should be consistent throughout the UI.